rust/chains/hyperlane-cosmos/src/providers/grpc.rs

```diff
@@ -147,7 +150,8 @@ impl WasmGrpcProvider {
 
         let auth_info = signer_info.auth_info(Fee::from_amount_and_gas(
             Coin::new(
-                Amount::from((gas_limit as f64 * DEFAULT_GAS_PRICE) as u64),
+                // The fee to pay is the gas limit * the gas price
+                (BigFloat::from(gas_limit) * self.gas_price()).try_into()?,
```

Comment:
Do you think this needs to be `(BigFloat::from(gas_limit) * self.gas_price()).ceil().try_into()?` ?

Iiuc it may be possible for
- gas_price to be the very minimum that the node accepts
- `BigFloat::from(gas_limit) * self.gas_price()` results in decimals, e.g. 100.1
- the try_into truncates this to 100, causing the amount to not be enough to cover the requested gas_limit at the required minimum gas price

Reply:
great point, I'm rounding up to the nearest integer now